### PR TITLE
soc: mcux: Add MKS22F12

### PR DESCRIPTION
**Pull request by rtalbott**
Adds support for the MKS22F12 line of mcu.

Signed-off-by: Rick Talbott <rtalbott@fastmail.com>

Depends on: https://github.com/zephyrproject-rtos/hal_nxp/pull/42

### CODE REVIEW

File: soc/arm/nxp_kinetis/ks2x/dts_fixup.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/*
```
Comment (galak) [inline soc/arm/nxp_kinetis/ks2x/dts_fixup.h:1]: This file has been phased out, you should be able to remove this.

**Review by galak — CHANGES_REQUESTED** [review]
Please remove dts_fixup.h.  

Any plans for a board port that utilizes this?  W/o one there isn't really any way to build test this functionality?

**Comment by galak** [comment]
I opened this issue https://github.com/zephyrproject-rtos/zephyr/issues/25153 for discussion around the idea of a buildtest board port.  I gave @rtalbott the idea for doing that since the MKS22F12 support in this PR doesn't have a board port to go with it.

File: soc/arm/nxp_kinetis/ks2x/dts_fixup.h (hunk 1 above)
Comment (rtalbott) [reply soc/arm/nxp_kinetis/ks2x/dts_fixup.h]: done

**Comment by rtalbott** [comment]
I've added a buildtest board.

**Comment by rtalbott** [comment]
I have had an issue getting USB working but I figured it out.  

The driver kinetis usb driver:  drivers/usb/device/usb_dc_kinetis.c
line 149 & 150:
	/* enable USB voltage regulator */
	//SIM->SOPT1 |= SIM_SOPT1_USBREGEN_MASK;

This is not supported in the MKS22.  Once commenting out, USB began to work.

Suggested solution is:
	/* enable USB voltage regulator */
#ifdef SIM_SOPT1_USBREGEN_MASK
	SIM->SOPT1 |= SIM_SOPT1_USBREGEN_MASK;
#endif

**Pull request closed without merging**